### PR TITLE
Restructure repo, refactor styling

### PR DESCRIPTION
**Pull request by anniero98**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
Somewhat analogous to https://github.com/calblueprint/dccentralkitchen/pull/24, but probably closer to the refactoring done last fall for the customer repo.

Pulled out the async Airtable functions into `lib/__Utils.js`, and cleaned up where I could. I fixed some bugs, deleted unused variables, and made functions take in args instead of using `this.state` so I could put them in the external files.

Refactored/deleted the extraneous css styling in the `styles` object of `styles.js`, then pulled out the existing components `styles.js` into `shared.js` and `checkout.js`.

Bugs:
- Checkout - # rewards applied per transaction needed to be added in addTransaction; basically for ACID properties. We also should track original points along with points earned, but not done in this PR.
- Login - as in the customer repo, need to set the first store's ID & name in the state upon component mount; otherwise, without interacting with the dropdown no storeId is available, and it crashes. Use case: Clerks logging in to the first store in the dropdown.

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Related PRs
Dependent on #6. New branch `annie/refactor` is dependent on this branch.

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review

Make sure everything works in a regular flow (transactions get created and linked). Check that there aren't extra `styles` objects lying around, or that other functions could be easily pulled out into `lib`

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps

Added a bunch of TODOs where there are things that could be cleaned up code-wise/edge cases are breaking the app, but I'm going to work on changing the AirTable calls structure first. I think I actually will import the AirTable schema generator and use it, etc, just because the `create__Data` Javascript-converting functions are pretty long and messy. And then I'm going to convert to `try/catch` with async/await (^:

Couple of things that I will probably fix along with that just cause....
- Add testing bypass (default clerk, default customer)
- Better error handling if clerk not found (don't reject in the initial Promise and crash, alert, or print on screen instead); currently any incorrect store name + password combo crashes the app
- We probably should add "original points" and "points awarded" (and discuss whether that should be called points earned)
- Make [transactionRecord], [itemRecord] (to get the first item in an array that's returned) .. not look like that? The linter doesn't like it, but it is an interesting shorthand.

Non-goals:
- A lot of `CheckoutScreen` seems to be refactor-able into components
- A lot of styling is still done inline, e.g `style={width: 50%}` 

Non-goals but nice-to-haves:
- Implement search for login and checkout (cart part)
- Refactor the way the phone number is taken in by having the parentheses be part of the input template (and do so in customer). Alternatively, refactor the formatting phone number and splicing portion of the code

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

**Breaking changes made with the AirTable columns and/or AirTable calls, so only once this branch is merged will it be usable again. But the whole flow works here

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC:  @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: screens/CheckoutScreen.js [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    const toSet =
+      filter == 'All'
+        ? this.state.fullProducts
+        : this.state.fullProducts.filter(product =>
+            product.category.includes(filter)
+          );
+    this.setState({ products: toSet });
+  };
+
+  // TODO why is this async
```
Comment (thumn) [inline screens/CheckoutScreen.js]: Wait yeah I forget why I made this a Promise...since I work with just state I think it's ok to remove the Promise...

File: screens/ClerkLoginScreen.js [hunk 2]
```diff
@@ -49,67 +28,42 @@ export default class ClerkLogin extends React.Component {
...
+    //   .then(storeRecord => {
+    //     if (storeRecord) {
+    //       let name = storeRecord['fields']['Store Name'];
+    //       this.setState({ storeName: name });
+    //     }
+    //   })
+    //   .catch(err => {
+    //     // TODO(thu): Make a more helpful error message.
+    //     console.error('Error retrieving store record from Airtable', err);
+    //   });
```
Comment (tommypoa) [inline screens/ClerkLoginScreen.js]: Why is this commented out?

**Comment by tommypoa** [comment]
Looks really good! Left a comment about the commented out part of the code, but otherwise functionality is working fine (after your AirTable fix).

File: screens/ClerkLoginScreen.js (hunk 2 above)
Comment (anniero98) [reply screens/ClerkLoginScreen.js]: I fixed!
